pkg/rlptools/wasm_utils.go

```diff
@@ -10,98 +10,100 @@ import (
 	istioclientgoextensionv1alpha1 "istio.io/client-go/pkg/apis/extensions/v1alpha1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	gatewayapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gatewayapiv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 
-	kuadrantv1beta1 "github.com/kuadrant/kuadrant-operator/api/v1beta1"
+	kuadrantv1beta2 "github.com/kuadrant/kuadrant-operator/api/v1beta2"
 	"github.com/kuadrant/kuadrant-operator/pkg/common"
 )
 
 var (
 	WASMFilterImageURL = common.FetchEnv("RELATED_IMAGE_WASMSHIM", "oci://quay.io/kuadrant/wasm-shim:latest")
+
+	PathMatchTypeMap = map[gatewayapiv1alpha2.PathMatchType]PatternOperator{
+		gatewayapiv1beta1.PathMatchExact:             PatternOperator(kuadrantv1beta2.EqualOperator),
+		gatewayapiv1beta1.PathMatchPathPrefix:        PatternOperator(kuadrantv1beta2.StartsWithOperator),
+		gatewayapiv1beta1.PathMatchRegularExpression: PatternOperator(kuadrantv1beta2.MatchesOperator),
+	}
 )
 
-type GatewayAction struct {
-	Configurations []kuadrantv1beta1.Configuration `json:"configurations"`
+type SelectorSpec struct {
```

Comment:
Should we move this or part of it to a common package?

Reply:
good idea. What do you propose?

* `github.com/kuadrant/kuadrant-operator/api/v1beta2` ?
* `github.com/kuadrant/kuadrant-operator/pkg/types` ?

Reply:
done, moved to `github.com/kuadrant/kuadrant-operator/pkg/rlptools/wasm`